client/dashboard/src/pages/logs/Logs.tsx

```diff
@@ -0,0 +1,532 @@
+import { Page } from "@/components/page-layout";
+import { Badge } from "@/components/ui/badge";
+import { SearchBar } from "@/components/ui/search-bar";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Sheet,
+  SheetContent,
+} from "@/components/ui/sheet";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { useListToolLogs } from "@gram/client/react-query";
+import { HTTPToolLog } from "@gram/client/models/components";
+import { Icon } from "@speakeasy-api/moonshine";
+import { Copy, ExternalLink, FileCode, SquareFunction } from "lucide-react";
+import { useState, useRef, useEffect } from "react";
+
+function StatusIcon({ isSuccess }: { isSuccess: boolean }) {
+  if (isSuccess) {
+    return (
+      <div style={{ color: "var(--fill-success-default, #5a8250)" }}>
+        <Icon name="check" className="items-center size-4" />
+      </div>
+    );
+  }
+  return (
+    <div style={{ color: "var(--fill-destructive-default, #c83228)" }}>
+      <Icon name="x" className="size-4" />
+    </div>
+  );
+}
+
+export default function Logs() {
+  const [searchQuery, setSearchQuery] = useState("");
```

Comment:
I would potentially represent these all as a single state object with

```ts
const filters = useState<{
  searchQuery: string | null;
  toolTypeFilter: string | null;
  serverNameFilter: string | null;
  statusFilter: string | null,
  selectedLog: string | null
}>({ searchQuery: null, toolTypeFilter: null, serverNameFilter: null, statusFilter: null, selectedLog: null})
```

you can also leave them as just strings. I like using explicit null but definitely not everyone likes that

Reply:
Done